libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java

```diff
@@ -168,42 +160,67 @@ public static synchronized String getRSAPublicString(String name, int length) {
      * @return RSA private key.
      * @deprecated Will be removed in Mobile SDK 8.0. Use {@link KeyStoreWrapper#getRSAPrivateKey(String, int)} instead.
      */
-    public static synchronized PrivateKey getRSAPrivateKey(String name, int length) {
+    public static PrivateKey getRSAPrivateKey(String name, int length) {
         return KeyStoreWrapper.getInstance().getRSAPrivateKey(name, length);
     }
 
-    private static void generateEncryptionKey(String name) {
+    /**
+     * Upgrades the keys stored in SharedPrefs to encrypted keys. This is a one-time
+     * migration step that's run while upgrading to Mobile SDK 7.1.
+     */
+    public synchronized static void upgradeTo7Dot1() {
+        final SharedPreferences prefs = SalesforceSDKManager.getInstance().getAppContext().getSharedPreferences(SHARED_PREF_FILE, 0);
+        final Map<String, ?> prefContents = prefs.getAll();
+        if (prefContents != null) {
+            final Set<String> keys = prefContents.keySet();
+            for (final String key : keys) {
+                if (key != null && key.startsWith("id")) {
+                    final String value = prefs.getString(key, null);
+                    if (value != null) {
+                        final PublicKey publicKey = KeyStoreWrapper.getInstance().getECPublicKey(KEYSTORE_ALIAS);
```

Comment:
Should it be `KeyStoreWrapper.getInstance().getRSAPublicKey`

Reply:
Nice catch. Too quick on the auto-complete.